nikola/plugins/compile/rest/__init__.py

```diff
@@ -81,14 +81,17 @@ def compile_string(self, data, source_path=None, is_two_file=True, post=None, la
             'language_code': LEGAL_VALUES['DOCUTILS_LOCALES'].get(LocaleBorg().current_lang, 'en')
         }
 
+        from nikola import shortcodes as sc
```

Comment:
What about other compilers? Also, I think it might be a slightly better-looking API to put `extract_shortcodes` in the Nikola object as well.